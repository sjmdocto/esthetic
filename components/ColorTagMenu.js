import React, {useState} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';
import Icon from 'react-native-vector-icons/Feather';

const colorKey = {
  black: 'Black',
  white: 'White',
  grey: 'Grey',
  red: 'Red',
  orange: 'Orange',
  yellow: 'Yellow',
  green: 'Green',
  blue: 'Blue',
  purple: 'Purple',
  multicolored: 'Multicolored',
};

const ColorTagMenu = (props) => {
  const [colorSelect, setColorSelect] = useState(colorKey.black);
  const colorSelectHandler = (color) => {
    setColorSelect(color);
    console.log(colorSelect);
  };
  return (
    <Menu>
      <MenuTrigger>
        <View style={styles.tagSelector}>
          <Icon name={'tag'} size={14} color={'black'} style={{right: 5}} />
          <Text>{colorSelect}</Text>
        </View>
      </MenuTrigger>
      <MenuOptions>
        <MenuOption
          value={1}
          text={colorKey.black}
          onSelect={colorSelectHandler.bind(this, colorKey.black)}
        />
        <MenuOption
          value={2}
          text={colorKey.white}
          onSelect={colorSelectHandler.bind(this, colorKey.white)}
        />
        <MenuOption
          value={3}
          text={colorKey.grey}
          onSelect={colorSelectHandler.bind(this, colorKey.grey)}
        />
        <MenuOption
          value={4}
          text={colorKey.red}
          onSelect={colorSelectHandler.bind(this, colorKey.red)}
        />
        <MenuOption
          value={5}
          text={colorKey.orange}
          onSelect={colorSelectHandler.bind(this, colorKey.orange)}
        />
        <MenuOption
          value={6}
          text={colorKey.yellow}
          onSelect={colorSelectHandler.bind(this, colorKey.yellow)}
        />
        <MenuOption
          value={7}
          text={colorKey.green}
          onSelect={colorSelectHandler.bind(this, colorKey.green)}
        />
        <MenuOption
          value={8}
          text={colorKey.blue}
          onSelect={colorSelectHandler.bind(this, colorKey.blue)}
        />
        <MenuOption
          value={9}
          text={colorKey.purple}
          onSelect={colorSelectHandler.bind(this, colorKey.purple)}
        />
        <MenuOption
          value={10}
          text={colorKey.multicolored}
          onSelect={colorSelectHandler.bind(this, colorKey.multicolored)}
        />
      </MenuOptions>
    </Menu>
  );
};

const styles = StyleSheet.create({
  tagSelector: {
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: 'white',
    marginHorizontal: 5,
    paddingVertical: 2,
    borderRadius: 9,
  },
});

export default ColorTagMenu;
