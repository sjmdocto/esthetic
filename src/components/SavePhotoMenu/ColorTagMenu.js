import React, {useState} from 'react';
import {View, Text} from 'react-native';
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';
import Icon from 'react-native-vector-icons/Feather';
import colorKey from '../../util/colorKey';
import {styles} from './TagMenu.style';

const colorTag = {
  black: 'Black',
  white: 'White',
  grey: 'Grey',
  brown: 'Brown',
  red: 'Red',
  orange: 'Orange',
  yellow: 'Yellow',
  green: 'Green',
  blue: 'Blue',
  purple: 'Purple',
  pink: 'Pink',
  multicolored: 'Multicolored',
};

/**
 * Popup menu for selecting color tag, called by SavePhotoMenu
 * @function ColorTagMenu
 * @param {*} props
 * @param {func} props.setColorTag
 */

const ColorTagMenu = (props) => {
  const [colorSelect, setColorSelect] = useState(colorTag.black);
  /**
   * Helper function for clothingSelectHandler.
   * @function tagToKey
   * @param {string} tag
   * @returns {void}
   */
  const tagToKey = (tag) => {
    switch (tag) {
      case 'Black':
        props.setColorTag(colorKey.black);
        break;
      case 'White':
        props.setColorTag(colorKey.white);
        break;
      case 'Grey':
        props.setColorTag(colorKey.grey);
        break;
      case 'Brown':
        props.setColorTag(colorKey.brown);
        break;
      case 'Red':
        props.setColorTag(colorKey.red);
        break;
      case 'Orange':
        props.setColorTag(colorKey.orange);
        break;
      case 'Yellow':
        props.setColorTag(colorKey.yellow);
        break;
      case 'Green':
        props.setColorTag(colorKey.green);
        break;
      case 'Blue':
        props.setColorTag(colorKey.blue);
        break;
      case 'Purple':
        props.setColorTag(colorKey.purple);
        break;
      case 'Pink':
        props.setColorTag(colorKey.pink);
        break;
      case 'Multicolored':
        props.setColorTag(colorKey.multicolored);
        break;
      default:
        console.warn('tagToKey error');
        break;
    }
  };
  /**
   * Handler for selecting a color tag from the menu
   * @function colorSelectHandler
   * @param {string} color
   * @returns {void}
   */
  const colorSelectHandler = (color) => {
    setColorSelect(color);
    tagToKey(color);
  };

  return (
    <Menu>
      <MenuTrigger
        customStyles={{triggerTouchable: {testID: 'ColorTagMenu-trigger'}}}>
        <View
          style={styles.tagSelector}
          accessible={true}
          accessibilityLabel={'Select color tag'}>
          <Icon name={'tag'} size={14} color={'black'} style={styles.icon} />
          <Text>{colorSelect}</Text>
        </View>
      </MenuTrigger>
      <MenuOptions>
        <MenuOption
          value={1}
          text={colorTag.black}
          onSelect={colorSelectHandler.bind(this, colorTag.black)}
        />
        <MenuOption
          value={2}
          text={colorTag.white}
          onSelect={colorSelectHandler.bind(this, colorTag.white)}
        />
        <MenuOption
          value={3}
          text={colorTag.grey}
          onSelect={colorSelectHandler.bind(this, colorTag.grey)}
        />
        <MenuOption
          value={4}
          text={colorTag.brown}
          onSelect={colorSelectHandler.bind(this, colorTag.brown)}
        />
        <MenuOption
          value={5}
          text={colorTag.red}
          onSelect={colorSelectHandler.bind(this, colorTag.red)}
        />
        <MenuOption
          value={6}
          text={colorTag.orange}
          onSelect={colorSelectHandler.bind(this, colorTag.orange)}
        />
        <MenuOption
          value={7}
          text={colorTag.yellow}
          onSelect={colorSelectHandler.bind(this, colorTag.yellow)}
        />
        <MenuOption
          value={8}
          text={colorTag.green}
          onSelect={colorSelectHandler.bind(this, colorTag.green)}
        />
        <MenuOption
          value={9}
          text={colorTag.blue}
          onSelect={colorSelectHandler.bind(this, colorTag.blue)}
        />
        <MenuOption
          value={10}
          text={colorTag.purple}
          onSelect={colorSelectHandler.bind(this, colorTag.purple)}
        />
        <MenuOption
          value={11}
          text={colorTag.pink}
          onSelect={colorSelectHandler.bind(this, colorTag.pink)}
        />
        <MenuOption
          value={12}
          text={colorTag.multicolored}
          onSelect={colorSelectHandler.bind(this, colorTag.multicolored)}
        />
      </MenuOptions>
    </Menu>
  );
};

export default ColorTagMenu;
