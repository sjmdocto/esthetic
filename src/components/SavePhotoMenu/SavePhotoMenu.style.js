import {StyleSheet} from 'react-native';
import common from '../../styles/common.style';
import theme from '../../styles/theme.style';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginHorizontal: 20,
    marginVertical: common.SCREENHEIGHT / 4,
    backgroundColor: '#1F1F1F',
    borderRadius: 40,
  },
  titleContainer: {
    flex: 1,
    paddingTop: 10,
  },
  body: {
    flex: 3,
    flexDirection: 'row',
    paddingHorizontal: 10,
  },
  imageContainer: {
    flex: 45,
    marginRight: 10,
    justifyContent: 'center',
  },
  buttonsContainer: {
    flex: 2,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    paddingHorizontal: 1,
  },
  titleText: {
    color: 'white',
    fontSize: 28,
  },
  tagsContainer: {
    flex: 55,
    justifyContent: 'space-evenly',
  },
  tagsText: {
    color: 'white',
    fontSize: 14,
  },
  discardButton: {
    flex: 1,
    marginHorizontal: 20,
    height: 50,
    width: 1,
    backgroundColor: '#141414',
    borderRadius: 15,
    alignItems: 'center',
    justifyContent: 'center',
  },
  saveButton: {
    flex: 1,
    marginHorizontal: 20,
    height: 50,
    width: 1,
    backgroundColor: theme.EST_ORANGE,
    borderRadius: 15,
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonText: {
    color: 'white',
    fontSize: 18,
  },
  tagSelector: {
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: 'white',
    marginHorizontal: 5,
    paddingVertical: 2,
    borderRadius: 9,
  },
  photo: {
    width: 140,
    height: 140,
    marginRight: 10,
  },
});
