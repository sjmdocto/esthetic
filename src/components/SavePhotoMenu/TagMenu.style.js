import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  tagSelector: {
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: 'white',
    marginHorizontal: 5,
    paddingVertical: 2,
    borderRadius: 9,
  },
  icon: {
    right: 5,
  },
});
