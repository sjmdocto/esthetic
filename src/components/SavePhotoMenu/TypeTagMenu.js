import React, {useState} from 'react';
import {View, Text} from 'react-native';
import {
  Menu,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';
import Icon from 'react-native-vector-icons/Feather';
import typeKey from '../../util/typeKey';
import {styles} from './TagMenu.style';

const typeTag = {
  outerwear: 'Outerwear',
  formalwear: 'Formal Wear',
  sweaters: 'Sweaters',
  hoodies: 'Hoodies',
  buttonUps: 'Button Ups',
  tees: 'Tees',
  pants: 'Pants',
  shorts: 'Shorts',
  activeWear: 'Activewear',
  shoes: 'Shoes',
};

/**
 * Popup menu for selecting type tag, called by SavePhotoMenu
 * @function TypeTagMenu
 * @param {*} props
 * @param {func} props.setTypeTag
 */

const TypeTagMenu = (props) => {
  const [clothingSelect, setClothingSelect] = useState(typeTag.outerwear);
  /**
   * Helper function for clothingSelectHandler
   * @function tagToKey
   * @param {string} tag
   * @returns {void}
   */
  const tagToKey = (tag) => {
    switch (tag) {
      case 'Outerwear':
        props.setTypeTag(typeKey.outerwear);
        break;
      case 'Formal Wear':
        props.setTypeTag(typeKey.formalwear);
        break;
      case 'Sweaters':
        props.setTypeTag(typeKey.sweaters);
        break;
      case 'Hoodies':
        props.setTypeTag(typeKey.hoodies);
        break;
      case 'Button Ups':
        props.setTypeTag(typeKey.buttonUps);
        break;
      case 'Tees':
        props.setTypeTag(typeKey.tees);
        break;
      case 'Pants':
        props.setTypeTag(typeKey.pants);
        break;
      case 'Shorts':
        props.setTypeTag(typeKey.shorts);
        break;
      case 'Activewear':
        props.setTypeTag(typeKey.activeWear);
        break;
      case 'Shoes':
        props.setTypeTag(typeKey.shoes);
        break;
      default:
        console.warn('tagToKey error');
        break;
    }
  };
  /**
   * Handler for selecting a type tag from the menu
   * @function clothingSelectHandler
   * @param {string} tag
   * @returns {void}
   */
  const clothingSelectHandler = (clothing) => {
    setClothingSelect(clothing);
    tagToKey(clothing);
  };

  return (
    <Menu>
      <MenuTrigger>
        <View
          style={styles.tagSelector}
          accessible={true}
          accessibilityLabel={'Select clothing type tag'}>
          <Icon name={'tag'} size={14} color={'black'} style={styles.icon} />
          <Text>{clothingSelect}</Text>
        </View>
      </MenuTrigger>
      <MenuOptions>
        <MenuOption
          value={1}
          text={typeTag.outerwear}
          onSelect={clothingSelectHandler.bind(this, typeTag.outerwear)}
        />
        <MenuOption
          value={2}
          text={typeTag.formalwear}
          onSelect={clothingSelectHandler.bind(this, typeTag.formalwear)}
        />
        <MenuOption
          value={3}
          text={typeTag.sweaters}
          onSelect={clothingSelectHandler.bind(this, typeTag.sweaters)}
        />
        <MenuOption
          value={4}
          text={typeTag.hoodies}
          onSelect={clothingSelectHandler.bind(this, typeTag.hoodies)}
        />
        <MenuOption
          value={5}
          text={typeTag.buttonUps}
          onSelect={clothingSelectHandler.bind(this, typeTag.buttonUps)}
        />
        <MenuOption
          value={6}
          text={typeTag.tees}
          onSelect={clothingSelectHandler.bind(this, typeTag.tees)}
        />
        <MenuOption
          value={7}
          text={typeTag.pants}
          onSelect={clothingSelectHandler.bind(this, typeTag.pants)}
        />
        <MenuOption
          value={8}
          text={typeTag.shorts}
          onSelect={clothingSelectHandler.bind(this, typeTag.shorts)}
        />
        <MenuOption
          value={9}
          text={typeTag.activeWear}
          onSelect={clothingSelectHandler.bind(this, typeTag.activeWear)}
        />
        <MenuOption
          value={10}
          text={typeTag.shoes}
          onSelect={clothingSelectHandler.bind(this, typeTag.shoes)}
        />
      </MenuOptions>
    </Menu>
  );
};

export default TypeTagMenu;
