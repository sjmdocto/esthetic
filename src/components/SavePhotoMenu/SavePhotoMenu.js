import React, {useState} from 'react';
import {View, Text, Modal, TouchableOpacity, Image} from 'react-native';
import {MenuProvider} from 'react-native-popup-menu';
import ColorTagMenu from './ColorTagMenu';
import TypeTagMenu from './TypeTagMenu';
import {styles} from './SavePhotoMenu.style';

/**
 * Popup menu for when you press the shutter
 * @function SavePhotoMenu
 * @param {*} props
 * @param {boolean} props.saveVisibility
 * @param {func} props.onSave
 * @param {func} props.onDiscard
 * @param {string} props.photo (in base64)
 * @param {array} props.closet
 */

const SavePhotoMenu = (props) => {
  const [colorTag, setColorTag] = useState(1);
  const [typeTag, setTypeTag] = useState(1);

  return (
    <Modal visible={props.saveVisibility} transparent={true}>
      <MenuProvider skipInstanceCheck={true}>
        <View
          style={styles.container}
          accessible={true}
          accessibilityLabel={'Save Photo Menu'}>
          <View style={styles.titleContainer}>
            <Text style={styles.titleText}>Select Tags</Text>
          </View>
          <View style={styles.body}>
            <View style={styles.imageContainer}>
              <Image
                style={styles.photo}
                source={{uri: `data:image/png;base64,${props.photo}`}}
              />
            </View>
            <View style={styles.tagsContainer}>
              <Text style={styles.tagsText}>Color:</Text>
              <ColorTagMenu setColorTag={setColorTag} />
              <Text style={styles.tagsText}>Type:</Text>
              <TypeTagMenu setTypeTag={setTypeTag} />
            </View>
          </View>
          <View style={styles.buttonsContainer}>
            <TouchableOpacity
              onPress={props.onDiscard}
              style={styles.discardButton}
              accessible={true}
              accessibilityLabel={'Discard button'}>
              <View>
                <Text style={styles.buttonText}>Discard</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity
              onPress={props.onSave.bind(this, colorTag, typeTag)}
              style={styles.saveButton}
              accessible={true}
              accessibilityLabel={'Save button'}>
              <View>
                <Text style={styles.buttonText}>Save</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </MenuProvider>
    </Modal>
  );
};

export default SavePhotoMenu;
