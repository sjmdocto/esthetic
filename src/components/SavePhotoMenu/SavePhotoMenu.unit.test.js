import React from 'react';
import ColorTagMenu from './ColorTagMenu';
import SavePhotoMenu from './SavePhotoMenu';
import TypeTagMenu from './TypeTagMenu';
import {fireEvent, render} from 'react-native-testing-library';
import {MenuProvider} from 'react-native-popup-menu';

describe('Save Photo Menu test suite:', () => {
  describe('ColorTagMenu tests: ', () => {
    const mockSetColorTag = jest.fn().mockName('setColorTag');
    test('Renders properly', () => {
      const {toJSON} = render(
        <MenuProvider>
          <ColorTagMenu setColorTag={mockSetColorTag} />
        </MenuProvider>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
    // NOTE:
    // react-native-popup-menu isn't friendly with testing,
    // so it's functionality can't be tested at the moment
    // test('Menu options work', () => {
    //   const {getByTestId, toJSON} = render(
    //     <MenuProvider>
    //       <ColorTagMenu setColorTag={mockSetColorTag} />
    //     </MenuProvider>,
    //   );
    //   fireEvent.press(getByTestId('ColorTagMenu-trigger'));
    //   fireEvent.press(getByTestId('option-1'));
    //   expect(toJSON()).toMatchSnapshot();
    // });
  });
  describe('TypeTagMenu tests: ', () => {
    const mockSetTypeTag = jest.fn().mockName('setTypeTag');
    test('Renders properly', () => {
      const {toJSON} = render(
        <MenuProvider>
          <TypeTagMenu setTypeTag={mockSetTypeTag} />
        </MenuProvider>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
  describe('SavePhotoMenu tests: ', () => {
    const mockOnDiscard = jest.fn().mockName('onDiscard');
    const mockOnSave = jest.fn().mockName('onSave');
    const testPhoto = 'base64';
    beforeEach(() => {
      mockOnDiscard.mockReset();
      mockOnSave.mockReset();
    });

    test('Renders properly', () => {
      const {toJSON} = render(
        <SavePhotoMenu onSave={mockOnSave} photo={testPhoto} />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    test('Discard button is functional', () => {
      const {getByLabelText} = render(
        <SavePhotoMenu
          onSave={mockOnSave}
          onDiscard={mockOnDiscard}
          photo={'a'}
        />,
      );
      fireEvent.press(getByLabelText('Discard button'));
      expect(mockOnDiscard).toHaveBeenCalled();
    });

    test('Save button is functional', () => {
      const {getByLabelText} = render(
        <SavePhotoMenu onSave={mockOnSave} photo={'a'} />,
      );
      fireEvent.press(getByLabelText('Save button'));
      expect(mockOnSave).toHaveBeenCalled();
    });
  });
});
