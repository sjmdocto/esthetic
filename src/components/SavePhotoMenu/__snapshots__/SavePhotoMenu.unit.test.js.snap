// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save Photo Menu test suite: ColorTagMenu tests:  Renders properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="ColorTagMenu-trigger"
        >
          <View
            ctx={
              Object {
                "menuActions": Object {
                  "_getOpenedMenu": [Function],
                  "_notify": [Function],
                  "closeMenu": [Function],
                  "isMenuOpen": [Function],
                  "openMenu": [Function],
                  "toggleMenu": [Function],
                },
                "menuRegistry": Object {
                  "getAll": [Function],
                  "getMenu": [Function],
                  "setOptionsCustomStyles": [Function],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                  "updateLayoutInfo": [Function],
                },
              }
            }
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessibilityLabel="Select color tag"
              accessible={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 9,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "paddingVertical": 2,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 14,
                    },
                    Object {
                      "right": 5,
                    },
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text>
                Black
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;

exports[`Save Photo Menu test suite: SavePhotoMenu tests:  Renders properly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessibilityLabel="Save Photo Menu"
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1F1F1F",
            "borderRadius": 40,
            "flex": 1,
            "marginHorizontal": 20,
            "marginVertical": 333.5,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "paddingTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 28,
              }
            }
          >
            Select Tags
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 3,
              "flexDirection": "row",
              "paddingHorizontal": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 45,
                "justifyContent": "center",
                "marginRight": 10,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "data:image/png;base64,base64",
                }
              }
              style={
                Object {
                  "height": 140,
                  "marginRight": 10,
                  "width": 140,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 55,
                "justifyContent": "space-evenly",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              Color:
            </Text>
            <View>
              <View
                collapsable={false}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  testID="ColorTagMenu-trigger"
                >
                  <View
                    ctx={
                      Object {
                        "menuActions": Object {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": Object {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      Array [
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityLabel="Select color tag"
                      accessible={true}
                      style={
                        Object {
                          "backgroundColor": "white",
                          "borderRadius": 9,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginHorizontal": 5,
                          "paddingVertical": 2,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 14,
                            },
                            Object {
                              "right": 5,
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                      <Text>
                        Black
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                }
              }
            >
              Type:
            </Text>
            <View>
              <View
                collapsable={false}
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    ctx={
                      Object {
                        "menuActions": Object {
                          "_getOpenedMenu": [Function],
                          "_notify": [Function],
                          "closeMenu": [Function],
                          "isMenuOpen": [Function],
                          "openMenu": [Function],
                          "toggleMenu": [Function],
                        },
                        "menuRegistry": Object {
                          "getAll": [Function],
                          "getMenu": [Function],
                          "setOptionsCustomStyles": [Function],
                          "subscribe": [Function],
                          "unsubscribe": [Function],
                          "updateLayoutInfo": [Function],
                        },
                      }
                    }
                    style={
                      Array [
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessibilityLabel="Select clothing type tag"
                      accessible={true}
                      style={
                        Object {
                          "backgroundColor": "white",
                          "borderRadius": 9,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginHorizontal": 5,
                          "paddingVertical": 2,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "black",
                              "fontSize": 14,
                            },
                            Object {
                              "right": 5,
                            },
                            Object {
                              "fontFamily": "Feather",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                      <Text>
                        Outerwear
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
              "paddingHorizontal": 1,
            }
          }
        >
          <View
            accessibilityLabel="Discard button"
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#141414",
                "borderRadius": 15,
                "flex": 1,
                "height": 50,
                "justifyContent": "center",
                "marginHorizontal": 20,
                "opacity": 1,
                "width": 1,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
              >
                Discard
              </Text>
            </View>
          </View>
          <View
            accessibilityLabel="Save button"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(227, 131, 4)",
                "borderRadius": 15,
                "flex": 1,
                "height": 50,
                "justifyContent": "center",
                "marginHorizontal": 20,
                "opacity": 1,
                "width": 1,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                  }
                }
              >
                Save
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      pointerEvents="box-none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RCTSafeAreaView>
  </View>
</Modal>
`;

exports[`Save Photo Menu test suite: TypeTagMenu tests:  Renders properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            ctx={
              Object {
                "menuActions": Object {
                  "_getOpenedMenu": [Function],
                  "_notify": [Function],
                  "closeMenu": [Function],
                  "isMenuOpen": [Function],
                  "openMenu": [Function],
                  "toggleMenu": [Function],
                },
                "menuRegistry": Object {
                  "getAll": [Function],
                  "getMenu": [Function],
                  "setOptionsCustomStyles": [Function],
                  "subscribe": [Function],
                  "unsubscribe": [Function],
                  "updateLayoutInfo": [Function],
                },
              }
            }
            style={
              Array [
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessibilityLabel="Select clothing type tag"
              accessible={true}
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 9,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginHorizontal": 5,
                  "paddingVertical": 2,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "black",
                      "fontSize": 14,
                    },
                    Object {
                      "right": 5,
                    },
                    Object {
                      "fontFamily": "Feather",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text>
                Outerwear
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </RCTSafeAreaView>
</View>
`;
